json_encoder.go

```diff
@@ -104,6 +105,18 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 // grade-school notation otherwise).
 func (enc *jsonEncoder) AddFloat64(key string, val float64) {
 	enc.addKey(key)
+	if math.IsNaN(val) {
+		enc.bytes = append(enc.bytes, `"NaN"`...)
+		return
+	}
+	if math.IsInf(val, 1) {
+		enc.bytes = append(enc.bytes, `"+Inf"`...)
+		return
+	}
+	if math.IsInf(val, -1) {
```

Comment:
Good point, I totally missed that we're doing different values for them. What you have is fine.